### PR TITLE
test(dmn): fix flaky EvaluateDecisionTest

### LINKED ISSUE #15662: EvaluateDecisionTest.shouldMapRequestAndResponse is flaky
**Summary**

- How often does the test fail? 
  - So far twice
- Does it block your work?
  - No
- Do we suspect that it is a real failure?
  - No

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
 Error:    Run 1: EvaluateDecisionTest.shouldMapRequestAndResponse:92 
expected: 1
 but was: 17
Error:    Run 2: EvaluateDecisionTest.shouldMapRequestAndResponse:92 
expected: 1
 but was: 19
Error:    Run 3: EvaluateDecisionTest.shouldMapRequestAndResponse:92 
expected: 1
 but was: 20
Error:    Run 4: EvaluateDecisionTest.shouldMapRequestAndResponse:92 
expected: 1
 but was: 21
</pre>
</details>

**Hypotheses**

Can't tell, not my area of expertise :upside_down_face: 

**Logs**

https://github.com/camunda/zeebe/actions/runs/7261544437/job/19782956153

### PR DESCRIPTION
**Pull request by megglos**
## Description

Test was referencing a static field that got eventually mutated, resulting int failing asserts on the evaluationCount value.

image

## Related issues

closes #15662

### CODE REVIEW

File: zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java [hunk 1]
```diff
@@ -66,9 +66,9 @@ public void registerWith(final StubbedBrokerClient gateway) {
   @Override
   public BrokerResponse<DecisionEvaluationRecord> handle(
       final BrokerEvaluateDecisionRequest request) throws Exception {
-    // In practise the decision instance key is generated newly by the broker
-    final long decisionInstanceKey = DECISION_RECORD.getDecisionKey() + 1;
-    return new BrokerResponse<>(DECISION_RECORD, request.getPartitionId(), decisionInstanceKey);
+    final DecisionEvaluationRecord evaluationRecord = createDecisionRecord();
+    return new BrokerResponse<>(
+        evaluationRecord, request.getPartitionId(), evaluationRecord.getDecisionKey());
```
Comment (tmetzke) [inline zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:71]: 💭 Previously, this effectively was `evaluationRecord.getDecisionKey() + 1`. I am still trying to understand why and if this needs to be adjusted here.

File: zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java [hunk 2]
```diff
@@ -66,9 +66,9 @@ public void registerWith(final StubbedBrokerClient gateway) {
   @Override
   public BrokerResponse<DecisionEvaluationRecord> handle(
       final BrokerEvaluateDecisionRequest request) throws Exception {
-    // In practise the decision instance key is generated newly by the broker
-    final long decisionInstanceKey = DECISION_RECORD.getDecisionKey() + 1;
```
Comment (megglos) [inline zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: I didn't get how this was relevant, it was test setup code that increased the key value which was then validated in the test, I would argue there is no point in testing test setup code? I thus removed the modification of the key here.

**Review by tmetzke — APPROVED** [review]
👍 Fantastic 🚀

File: zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java (hunk 2 above)
Comment (tmetzke) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: That was also my impression a bit. We can also ask @koevskinikola if he still knows 🙂

Comment (megglos) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: Found that piece was added by @korthout on main recently https://github.com/camunda/zeebe/commit/164c854c484317084ae2e801d7c9b9c4bfb99a0c let's sync on whether we wanna add it back.

Comment (tmetzke) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: Ah, I see, good catch 👍 I think we should re-add this then, yeah.

Comment (megglos) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: Would like to understand the reason for stub code that increments the key to be validated by the test still.

Comment (korthout) [reply zeebe/gateway/src/test/java/io/camunda/zeebe/gateway/api/decision/EvaluateDecisionStub.java:70]: Each decision evaluation receives a unique key from the broker. This key is different from the decision key, which uniquely identifies the decision to evaluate. Rather it is the unique evaluation (or instance) of that decision.

This key is responded back to the client so the user can use it to find the decision instance (the evaluated decision) in Operate. 

The test doesn't need to be this specific to pass, but it provides a more realistic test case. It's just more correct semantically.

**Pull request merged**